[bug report] undefined: os.UserHomeDir

<!-- Please use this template while reporting a bug and provide as much info as possible. Thanks!-->


**What happened**:
go build failed

```
# github.com/kubeedge/kubeedge/vendor/k8s.io/kubernetes/pkg/credentialprovider
../vendor/k8s.io/kubernetes/pkg/credentialprovider/config.go:63:22: undefined: os.UserHomeDir

make[1]: *** [edge_core] Interrupt
```
**What you expected to happen**:


**How to reproduce it (as minimally and precisely as possible)**:
```
# cd edge
# make

go build cmd/edge_core.go
# github.com/kubeedge/kubeedge/vendor/k8s.io/kubernetes/pkg/credentialprovider
../vendor/k8s.io/kubernetes/pkg/credentialprovider/config.go:63:22: undefined: os.UserHomeDir
make: *** [edge_core] Error 2
```

**Anything else we need to know?**:

**Environment**:
- KubeEdge version:
master 
- Hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Others:
go version:
go version go1.11.12 linux/amd64



do we need go 1.12 ??

--- a/docs/setup/setup.md
+++ b/docs/setup/setup.md
@@ -18,7 +18,7 @@
     - --insecure-bind-address=0.0.0.0
     ```
 
-+ **Go** The minimum required go version is 1.11. You can install this version by using [this website.](https://golang.org/dl/) 
++ **Go** The minimum required go version is 1.12. You can install this version by using [this website.](https://golang.org/dl/) 
 
 + (**Optional**)KubeEdge also supports https connection to Kubernetes apiserver. Follow the steps in [Kubernetes Documentation](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/) to create the kubeconfig file.
 

